File: Segundo/PDOO/CivitasJava/src/main/java/civitas/CasillaSorpresa.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package civitas;

import java.util.ArrayList;

/**
 *
 * @author jmramirez
 */
public class CasillaSorpresa extends Casilla {
    
    private MazoSorpresas refMS;
    
    CasillaSorpresa(MazoSorpresas mazo, String n) {
        super(n);
        refMS = mazo;
    }
    
    void recibeJugador(int actual, ArrayList<Jugador> todos) {
        if (jugadorCorrecto(actual, todos)) {
            informe(actual, todos);
            var siguiente = refMS.siguiente();
            siguiente.aplicarAJugador(actual, todos);
        }
    }
    
    public String toString() {
        String ret = "la casilla ";
        ret += this.nombre;
        ret += ", de tipo Sorpresa.";
        return ret;
    }
}

File: Segundo/PDOO/CivitasRuby/Civitas/CasillaSorpresa.rb
# encoding:utf-8

require_relative "MazoSorpresas"

module Civitas

    class CasillaSorpresa < Casilla

        def initialize(mazoSor, n)
            super(n)
            @mazo = mazoSor
        end

        def informe(actual, todos)
            Diario.instance.ocurre_evento("El jugador " + todos.at(actual).nombre + " ha caído en una casilla de tipo Sorpresa.")
        end

        def recibe_jugador(actual, todos)
            if jugadorCorrecto(actual, todos)
                informe(actual, todos)
                sorpresa = @mazo.siguiente()
                sorpresa.aplicarAJugador(actual, todos)
            end
        end

        def toString
            return "La casilla actual es " + @nombre + " y es de tipo Sorpresa.\n"
        end

    end

end